rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -76,12 +76,43 @@ void rsx::thread::send_event(u64 data1, u64 event_flags, u64 data3) const
 		error = sys_event_port_send(rsx_event_port, data1, event_flags, data3);
 	}
 
-	if (error && error + 0u != CELL_ENOTCONN)
+	if (!Emu.IsPaused() && error && error + 0u != CELL_ENOTCONN)
 	{
 		fmt::throw_exception("rsx::thread::send_event() Failed to send event! (error=%x)", +error);
 	}
 }
 
+std::unique_lock<shared_mutex> lock_sys_rsx_mutex()
+{
+	using lock_t = std::unique_lock<shared_mutex>;
+
+	const auto render = rsx::get_current_renderer();
+
+	if (!render)
+	{
+		return lock_t{};
+	}
+
+	return lock_t{render->sys_rsx_mtx};
+}
+
+void signal_gcm_intr_thread_offline(u32 queue_id)
```

Comment:
When all the code has the same prefix, it needs a refactor. The logic here belongs inside the 'render' object.